From 3df700360682a3944f95e32044c7e22f1ddda4a6 Mon Sep 17 00:00:00 2001
From: Sam Leffler <sleffler@chromium.org>
Date: Tue, 31 Jan 2012 11:03:31 -0800
Subject: [PATCH] CHROMIUMOS: wpa_debug: add syslog support for
 wpa_hexdump_ascii

Add missing support for wpa_hexdump_ascii with syslog.  This
actually cheats and just uses wpa_hexdump since the ascii
formatting is painful and unlikely to be desired with syslog.

BUG=chromium-os:25482
TEST=enable debuging and associate to Google-A; verify hexdumps are included in the log [tested with log to stdout and to syslog]

Change-Id: I47029c5660aa73141b905e3ce0e42355860d0941
---
 src/utils/wpa_debug.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/utils/wpa_debug.c b/src/utils/wpa_debug.c
index 68cbace..e6fc15a 100644
--- a/src/utils/wpa_debug.c
+++ b/src/utils/wpa_debug.c
@@ -418,6 +418,14 @@ static void _wpa_hexdump_ascii(int level, const char *title, const void *buf,
 #ifdef CONFIG_ANDROID_LOG
 	_wpa_hexdump(level, title, buf, len, show);
 #else /* CONFIG_ANDROID_LOG */
+
+#ifdef CONFIG_DEBUG_SYSLOG
+	if (wpa_debug_syslog) {
+		/* punt, too hard to deal with formatting */
+		_wpa_hexdump(level, title, buf, len, show);
+		return;
+	}
+#endif
 	wpa_debug_print_timestamp();
 #ifdef CONFIG_DEBUG_FILE
 	if (out_file) {
-- 
2.2.0.rc0.207.ga3a616c

